
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>ChIP-seq analysis</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/better.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="lcdb-wf" href="lcdb-wf.html" />
    <link rel="prev" title="Bulk RNA-seq analysis" href="rnaseq.html" />


  <link href="https://fonts.googleapis.com/css?family=Lato|Lato:bold,italic,bolditalic|Lora|Oswald|Inconsolata&display=swap" rel="stylesheet">
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
      <link rel="stylesheet" href="_static/style.css" type="text/css" />
  </head><body>
    <header id="pageheader"><h1><a href="index.html ">
        
    </a></h1></header>
  <div class="related top">
  <nav id="rellinks">
    <ul>
        <li>
          &larr;
          <a href="rnaseq.html" title="Previous document">Bulk RNA-seq analysis</a>
        </li>
        <li>
          <a href="lcdb-wf.html" title="Next document"><code class="docutils literal notranslate"><span class="pre">lcdb-wf</span></code></a>
          &rarr;
        </li>
    </ul>
  </nav>
  <nav id="breadcrumbs">
    <ul>
      <li><a href="index.html">Home</a></li> 
    </ul>
  </nav>
  </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="chip-seq-analysis">
<h1>ChIP-seq analysis<a class="headerlink" href="#chip-seq-analysis" title="Permalink to this heading">¶</a></h1>
<p>In comparison to RNA-seq analysis, ChIP-seq analysis can be a bit more
complicated. The experimental protocol itself is more complex, and has
additional variability imposed by antibody effectiveness.</p>
<p>Fundamentally, ChIP-seq is about finding where on chromatin a protein of
interest binds. To do this, we generally convert aligned reads into signal over
the genome, and wherever there is a large pileup of reads, we call that as
a “peak” and infer that the protein was found there.</p>
<p>This aspect of peak-calling can also be used with other chromatin-related
assays like ATAC-seq and CUT&amp;RUN albeit with some minor modifications. In
general, peak-calling is more of an art than a science. Multiple peak-calling
algorithms will often give different results, so a big part of ChIP-seq
analysis is understanding which algorithms are most appropriate for your
particular experiment and making well-justified decisions about which peaks to
use as the “final” set.</p>
<p><strong>The best one-stop-shop</strong> for learning about ChIP-seq analysis is the Harvard
Bioinformatics Core’s <a class="reference external" href="https://hbctraining.github.io/Intro-to-ChIPseq/schedule/2-day.html">Introduction to ChIP-seq using high-performance
computing</a> starts
with an overview of ChIP-seq, and continues through working with Linux, project
organization, alignment, QC, peak-calling, and comparing peaks between conditions.</p>
<p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S1046202320300591">Nakato 2021</a> has
a high-level overview of ChIP-seq, and may be more succinct than the HBC
workshop linked above.</p>
<section id="technical-details">
<h2>Technical details<a class="headerlink" href="#technical-details" title="Permalink to this heading">¶</a></h2>
<p>The current ENCODE guidelines for ChIP-seq can be found <a class="reference external" href="https://www.encodeproject.org/documents/ceb172ef-7474-4cd6-bfd2-5e8e6e38592e/&#64;&#64;download/attachment/ChIP-seq_ENCODE3_v3.0.pdf">here</a>.
Here are some highlights:</p>
<ul class="simple">
<li><p>two or more biological replicates</p></li>
<li><p>for mammalian genomes, 20 million usable fragments for narrow peaks (a
fragment is either one read for single-end sequencing, or one pair for
paired-end sequencing)</p></li>
<li><p>for mammalian genomes, 45 million usable fragments for broad peaks</p></li>
</ul>
<p>Input or H3 as a control? <a class="reference external" href="https://www.frontiersin.org/articles/10.3389/fgene.2014.00329/full">Flensburg 2014</a> compares
the use of whole-cell extract (input) vs H3 as a control for histone
modification ChIP-seq. They found that the called peaks are mostly similar no
matter which control is used. As you might expect, when using H3 there is
a depletion of signal around the TSS due to expected nucleosome depletion.
While there may be corner cases where the particular biological questions may
cause one to be more biologically meaningful, in general using input for
histone modifications seems well-justified.</p>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">ChIP-seq analysis</a><ul>
<li><a class="reference internal" href="#technical-details">Technical details</a></li>
</ul>
</li>
</ul>

  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/chipseq.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<form class="search" action="search.html" method="get">
  <input type="text" name="q"
   placeholder="type to search" />
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
  <div class="related bottom">
  <nav id="rellinks">
    <ul>
        <li>
          &larr;
          <a href="rnaseq.html" title="Previous document">Bulk RNA-seq analysis</a>
        </li>
        <li>
          <a href="lcdb-wf.html" title="Next document"><code class="docutils literal notranslate"><span class="pre">lcdb-wf</span></code></a>
          &rarr;
        </li>
    </ul>
  </nav>
  <nav id="breadcrumbs">
    <ul>
      <li><a href="index.html">Home</a></li> 
    </ul>
  </nav>
  </div>
  <footer id="pagefooter">&copy; 2020, Ryan Dale.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a>
      6.1.3.

  </footer>

  
  </body>
</html>