
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>Galaxy</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/better.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Emacs" href="emacs.html" />
    <link rel="prev" title="HTML and websites" href="html.html" />


  <link href="https://fonts.googleapis.com/css?family=Lato|Lato:bold,italic,bolditalic|Lora|Oswald|Inconsolata&display=swap" rel="stylesheet">
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
      <link rel="stylesheet" href="_static/style.css" type="text/css" />
  </head><body>
    <header id="pageheader"><h1><a href="index.html ">
        
    </a></h1></header>
  <div class="related top">
  <nav id="rellinks">
    <ul>
        <li>
          &larr;
          <a href="html.html" title="Previous document">HTML and websites</a>
        </li>
        <li>
          <a href="emacs.html" title="Next document">Emacs</a>
          &rarr;
        </li>
    </ul>
  </nav>
  <nav id="breadcrumbs">
    <ul>
      <li><a href="index.html">Home</a></li> 
    </ul>
  </nav>
  </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="galaxy">
<span id="id1"></span><h1>Galaxy<a class="headerlink" href="#galaxy" title="Permalink to this heading">¶</a></h1>
<a class="reference internal image-reference" href="_images/galaxyLogo.jpeg"><img alt="_images/galaxyLogo.jpeg" src="_images/galaxyLogo.jpeg" style="width: 50%;" /></a>
<p><a class="reference external" href="https://usegalaxy.org/">Galaxy</a> is an open-source web application built for scientific analyses featuring a public server located within the cloud, accessible open-source software packages, custom Galaxy tools (Galaxy ToolShed), and a user-friendly drag-and-drop interface. The platform is easy to learn, requiring no programming experience, and used to build reproducible bioinformatics workflows. Galaxy’s community also serves as one of the few active forums for bioinformaticians to collaborate, with over 150 active members monthly.</p>
<p>Multiple tools are available to the user, including popular ones such as FastQC (quality control), Bowtie2 (aligner), featureCounts (count reads in genes), and many more. Galaxy’s built-in drag-and-drop workflow also serves as a transparent way to quickly communicate analyses to other researchers and collaborators.</p>
<p>Galaxy is great for genomic analyses, especially for beginners or users without lots of computational experience. Overall, it’s one of the few cloud-based community hubs for bioinformaticians of all skill-levels. The Galaxy community is only getting stronger - with over 124,000 registered users since its inception in 2007 and over 2000 GalaxyHelp members total.</p>
<figure class="align-default" id="id3">
<img alt="_images/galaxyBulkRNASeqWF.png" src="_images/galaxyBulkRNASeqWF.png" />
<figcaption>
<p><span class="caption-text">The Galaxy UI. The lefthand sidebar shows the tools available to the user. The center features a workflow for Bulk RNASeq. The righthand sidebar shows additional information.</span><a class="headerlink" href="#id3" title="Permalink to this image">¶</a></p>
</figcaption>
</figure>
<section id="getting-started-with-galaxy">
<h2>Getting Started with Galaxy<a class="headerlink" href="#getting-started-with-galaxy" title="Permalink to this heading">¶</a></h2>
<p>Utilize the <a class="reference external" href="https://training.galaxyproject.org/training-material/">Galaxy Training pages</a> to understand what workflows are possible for Galaxy and a specific workflow is the <a class="reference external" href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#data-upload">Bulk RNA Seq Analysis</a> Galaxy tutorial</p>
</section>
<section id="benefits">
<h2>Benefits<a class="headerlink" href="#benefits" title="Permalink to this heading">¶</a></h2>
<ol class="arabic simple">
<li><p><strong>Easy to use</strong> A drag-and-drop interface allows users to quickly understand where the data is located and what is happening every step of the workflow. This helps really helps both those who are in and outside of the lab. Anybody can quickly understand the input and output of a workflow.</p></li>
<li><p><strong>Cloud-based server</strong> This enables bioinformaticians to quickly save and store information without the fear of it being lost on some random computer’s filesystem or stored on an accidentally corrupted USB. Workflows and stored data can be shared via URL links.</p></li>
<li><p><strong>Many tools available</strong> The bioinformatics field constantly pushes out novel tools, but at a rate that might be too difficult to keep up with. Galaxy’s tools provides an easy search function to find these tools at their most up-to-date version. A bioinformatician who previously was unaware of a tool might discover it and realize it’s a much better implementation, thus improving their workflow’s analysis. This is a great way for people to distribute their tools and have people actually use them.</p></li>
<li><p><strong>Galaxy’s training docs are well maintained and applicable</strong> The Galaxy community shines in making training for applied projects. The documents are interactive, often featuring video, slides, transcript, written tutorial, etc. The training documents provide a great way to understand the extent that Galaxy can help with your project.</p></li>
<li><p><strong>Galaxy is constantly being updated to be more user-friendly and advanced</strong> Not only does it have workflow management, Galaxy also boasts many tools for data visualization and even a Jupyter Notebook feature.</p></li>
</ol>
</section>
<section id="drawbacks">
<h2>Drawbacks<a class="headerlink" href="#drawbacks" title="Permalink to this heading">¶</a></h2>
<ol class="arabic simple">
<li><p><strong>Possible lack of customization</strong> Some tools may not include all options that the tool would offer if manually installed. For example, “Operate on Genomic Intervals:Intersect” and “BED:bedtools Intersect intervals” are the same tool in Galaxy yet <code class="docutils literal notranslate"><span class="pre">bedtools</span></code> the command-line tool offers more options for the user to set.</p></li>
<li><p><strong>Confusing documentation</strong> Some tools only have a link to the tool’s original documentation. This may be difficult to understand and apply to Galaxy since that documentation is often written with the command-line user in mind. A great example of a well documented tool is “Download and Extract reads in FASTA/Q”. It is a simple tool with understandable documentation. It also has flags for what Galaxy is internally doing and how Galaxy will output these files to the user.</p></li>
<li><p><strong>Lots of tools</strong> The naming of the tools splits between Galaxy-programmed tools and Galaxy-adapted tools created by other pre-existing packages can be a source of confustion. For example, in the “Get Data” tab, there are two options to get data from NCBI SRA that look very similar: “Faster Download and Extract Reads in FASTQ” and “Download and Extract Reads in FASTA/Q”.</p></li>
</ol>
</section>
<section id="papers">
<h2>Papers<a class="headerlink" href="#papers" title="Permalink to this heading">¶</a></h2>
<p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6030816/">The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update (Afghan, et al. 2018)</a></p>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">Galaxy</a><ul>
<li><a class="reference internal" href="#getting-started-with-galaxy">Getting Started with Galaxy</a></li>
<li><a class="reference internal" href="#benefits">Benefits</a></li>
<li><a class="reference internal" href="#drawbacks">Drawbacks</a></li>
<li><a class="reference internal" href="#papers">Papers</a></li>
</ul>
</li>
</ul>

  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/galaxy.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<form class="search" action="search.html" method="get">
  <input type="text" name="q"
   placeholder="type to search" />
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
  <div class="related bottom">
  <nav id="rellinks">
    <ul>
        <li>
          &larr;
          <a href="html.html" title="Previous document">HTML and websites</a>
        </li>
        <li>
          <a href="emacs.html" title="Next document">Emacs</a>
          &rarr;
        </li>
    </ul>
  </nav>
  <nav id="breadcrumbs">
    <ul>
      <li><a href="index.html">Home</a></li> 
    </ul>
  </nav>
  </div>
  <footer id="pagefooter">&copy; 2020, Ryan Dale.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a>
      5.3.0.

  </footer>

  
  </body>
</html>