
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>Bulk RNA-seq analysis</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/better.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ChIP-seq analysis" href="chipseq.html" />
    <link rel="prev" title="Variant calling" href="variant-calling.html" />


  <link href="https://fonts.googleapis.com/css?family=Lato|Lato:bold,italic,bolditalic|Lora|Oswald|Inconsolata&display=swap" rel="stylesheet">
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
      <link rel="stylesheet" href="_static/style.css" type="text/css" />
  </head><body>
    <header id="pageheader"><h1><a href="index.html ">
        
    </a></h1></header>
  <div class="related top">
  <nav id="rellinks">
    <ul>
        <li>
          &larr;
          <a href="variant-calling.html" title="Previous document">Variant calling</a>
        </li>
        <li>
          <a href="chipseq.html" title="Next document">ChIP-seq analysis</a>
          &rarr;
        </li>
    </ul>
  </nav>
  <nav id="breadcrumbs">
    <ul>
      <li><a href="index.html">Home</a></li> 
    </ul>
  </nav>
  </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="bulk-rna-seq-analysis">
<h1>Bulk RNA-seq analysis<a class="headerlink" href="#bulk-rna-seq-analysis" title="Permalink to this heading">¶</a></h1>
<p>RNA-seq measures what genes are transcribed. We typically differentiate between
bulk RNA-seq (measuring transcripts from many cells mixed together) and
single-cell RNA-seq (keeping track of which transcript measurements came from
which cell). This page is about bulk RNA-seq.</p>
<section id="harvard-bioinformatics-core-training-materials">
<h2>Harvard Bioinformatics Core training materials<a class="headerlink" href="#harvard-bioinformatics-core-training-materials" title="Permalink to this heading">¶</a></h2>
<p>These tutorials are the best way to learn RNA-seq:</p>
<ul class="simple">
<li><p><strong>TUTORIAL 1</strong>: <a class="reference external" href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon-flipped/schedule/links-to-lessons.html">Introduction to (bulk) RNA-seq using High-Performance Computing</a>. This walks you through the primary processing, mostly using command-line tools.</p></li>
<li><p><strong>TUTORIAL 2</strong>: <a class="reference external" href="https://hbctraining.github.io/DGE_workshop_salmon_online/schedule/links-to-lessons.html">Introduction to Differential Gene Expression Analysis</a>. This walks you through differential expression analysis in R.</p></li>
</ul>
<p>Note that these have prerequisites of <a class="reference external" href="https://hbctraining.github.io/Intro-to-shell-flipped/schedule/links-to-lessons.html">Introduction to shell</a>
and <a class="reference external" href="https://hbctraining.github.io/Intro-to-R-flipped/schedules/links-to-lessons.html">Introduction to
R</a>
respectively.</p>
<p>Upon completing those two tutorials, you will be able to run arbitrary RNA-seq
analyses, and you may be ready to learn Snakemake and either write your own
workflows or use something like <a class="reference external" href="https://github.com/lcdb/lcdb-wf">lcdb-wf</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The Harvard Bioinformatics Core tutorials sometimes change locations or get
updated or split. The above links are checked periodically for correctness,
but if you can’t find the linked tutorials, look around on their
<a class="reference external" href="https://github.com/hbctraining/main">main training page</a>.</p>
<p>These tutorials make some assumptions about using Harvard’s
high-performance compute cluster, O2. There are commands related to
installing or activating software that you will need to adapt to your local
HPC or other computing environment.</p>
<p>We have not rewritten these tutorials for NIH’s Biowulf, for example.</p>
</div>
</section>
<section id="skills-for-rna-seq">
<h2>Skills for RNA-seq<a class="headerlink" href="#skills-for-rna-seq" title="Permalink to this heading">¶</a></h2>
<p>This gives a very rough sense of what beginner/intermediate RNA-seq skills
might look like:</p>
<section id="level-1">
<h3>Level 1<a class="headerlink" href="#level-1" title="Permalink to this heading">¶</a></h3>
<ul class="simple">
<li><p>Be able to describe the lines and fields of the following formats: FASTQ, FASTA, BAM, GTF</p></li>
<li><p>Be able to compare and contrast the same formats (what information is
included/missing; how are they created; when you would use them)</p></li>
<li><p>Locate and prepare reference data (including aligner/pseudoaligner indexes)</p></li>
<li><p>Run FastQC on FASTQ and BAM files</p></li>
<li><p>Align reads with STAR, HISAT2, or other aligner and quantify reads with
featureCounts</p></li>
<li><p>Quantify reads with Kallisto or Salmon</p></li>
<li><p>Import a counts table into R</p></li>
<li><p>Run basic 2-condition DESeq2 differential expression analysis</p></li>
<li><p>Find the most highly-changed genes</p></li>
<li><p>Plot an MA plot or volcano plot</p></li>
<li><p>Export results to TSV or Excel</p></li>
<li><p>Do functional enrichment analysis to find pathways enriched in changed genes</p></li>
</ul>
</section>
<section id="level-2">
<h3>Level 2<a class="headerlink" href="#level-2" title="Permalink to this heading">¶</a></h3>
<ul class="simple">
<li><p>Run QC using multiple tools (preseq, rRNA contamination, Picard, RSeQC, MultiQC)</p></li>
<li><p>Interpret QC reports to make suggestions for future bench work</p></li>
<li><p>Interpret raw p-value histograms</p></li>
<li><p>Be able to describe what fold-change shrinkage is doing</p></li>
<li><p>Be able to describe how DESeq2 handles low counts</p></li>
<li><p>Work with more complex experimental designs (batch effects, interaction
terms) in DESeq2 and explain the results</p></li>
<li><p>Visualize data (bam, bigwig) in a genome browser</p></li>
<li><p>Using automated reproducible workflows (Snakemake, Nextflow, etc)</p></li>
</ul>
</section>
</section>
<section id="other-resources">
<h2>Other resources<a class="headerlink" href="#other-resources" title="Permalink to this heading">¶</a></h2>
<p>The remainder of this page goes into some more detail on various aspects of
RNA-seq analysis, to be used as supplemental material.</p>
<ul class="simple">
<li><p>The introductory paper <a href="#id3"><span class="problematic" id="id4">`Hitchiker's guide to expression analysis https://www.annualreviews.org/doi/10.1146/annurev-biodatasci-072018-021255&gt;`_</span></a> is co-authored
by many big names in the field and gives a great overview and history.</p></li>
<li><p>The <a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">DESeq2 paper</a>,
while going over the details of the popular differential expression
algorithm, is very approachable even for someone with not a lot of
math/stats/algorithm background.</p></li>
<li><p>If you’ve done the tutorials above, you will have seen the <a class="reference external" href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon-flipped/lessons/02_experimental_planning_considerations.html">experimental
planning considerations</a>
section from HBC training, which is a very good discussion on how much to
sequence (more samples or more depth?), how to avoid a confounded experiment,
and preventing batch effects in general.</p></li>
<li><p>Especially when doing <cite>in vitro</cite> research with cell lines, it’s important to
think about what a replicate really is. <a class="reference external" href="https://paasp.net/accurate-design-of-in-vitro-experiments-why-does-it-matter/">This blog post</a>
is a good discussion of the difference between technical replicates and
biological replicates <cite>in vitro</cite>.</p></li>
<li><p>A library may be stranded, reverse stranded, or unstranded depending on what
kit was used for the library prep. <a class="reference external" href="https://github.com/igordot/genomics/blob/master/notes/rna-seq-strand.md">These figures</a>
help visualize the different strand-specific protocols. If you’re unsure,
RSeQC’s <a class="reference external" href="http://rseqc.sourceforge.net/#infer-experiment-py">infer_experiment.py</a> can help you figure it
out given a BAM and a BED file of genes.</p></li>
<li><p>We base part of our RNA-seq template off of the <a class="reference external" href="https://www.bioconductor.org/packages/devel/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html">Bioconductor RNA-seq
workflow</a>, which shows all the steps of RNA-seq from within R.</p></li>
<li><p>In BSPC, we develop and maintain <a class="reference external" href="https://github.com/lcdb/lcdb-wf">lcdb-wf</a>, which automates much of RNA-seq, and
provides an extensive RMarkdown file for downstream analysis.</p></li>
<li><p>The <a class="reference external" href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2 vignette</a>
is the authoritative source on how to use DESeq2.</p></li>
<li><p>The <a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">DESeq2 paper</a>
is very well written, and describes how DESeq2 is actually working</p></li>
<li><p>A nice treatment of <a class="reference external" href="http://genomicsclass.github.io/book/pages/interactions_and_contrasts.html">interaction terms</a>,
along with plots to help understand what’s being tested.</p></li>
<li><p>For complex experimental designs, this <a class="reference external" href="https://github.com/tavareshugo/tutorial_DESeq2_contrasts/blob/main/DESeq2_contrasts.md">tutorial</a>
shows an elegant, general method for creating the proper contrasts.</p></li>
<li><p>You may have heard the terms RPKM, FPKM, RPM, and TPM. Which to use? Short
answer: use TPM. Longer answer, with figures, is here:
<a class="reference external" href="https://ro-che.info/articles/2016-11-28-rna-seq-normalization">https://ro-che.info/articles/2016-11-28-rna-seq-normalization</a>. The
accompanying <a class="reference external" href="https://ro-che.info/docs/2016-11-27-rna-seq.pdf">slides</a> are
useful for discussion.</p></li>
</ul>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">Bulk RNA-seq analysis</a><ul>
<li><a class="reference internal" href="#harvard-bioinformatics-core-training-materials">Harvard Bioinformatics Core training materials</a></li>
<li><a class="reference internal" href="#skills-for-rna-seq">Skills for RNA-seq</a><ul>
<li><a class="reference internal" href="#level-1">Level 1</a></li>
<li><a class="reference internal" href="#level-2">Level 2</a></li>
</ul>
</li>
<li><a class="reference internal" href="#other-resources">Other resources</a></li>
</ul>
</li>
</ul>

  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/rnaseq.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<form class="search" action="search.html" method="get">
  <input type="text" name="q"
   placeholder="type to search" />
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
  <div class="related bottom">
  <nav id="rellinks">
    <ul>
        <li>
          &larr;
          <a href="variant-calling.html" title="Previous document">Variant calling</a>
        </li>
        <li>
          <a href="chipseq.html" title="Next document">ChIP-seq analysis</a>
          &rarr;
        </li>
    </ul>
  </nav>
  <nav id="breadcrumbs">
    <ul>
      <li><a href="index.html">Home</a></li> 
    </ul>
  </nav>
  </div>
  <footer id="pagefooter">&copy; 2020, Ryan Dale.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a>
      6.1.3.

  </footer>

  
  </body>
</html>